File: src/main/java/com/ultreon/bubbles/util/system/OS.java
package com.ultreon.bubbles.util.system;

import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class OS {
    private final String name;
    private final String version;

    public OS() {
        this(System.getProperty("os.name"), System.getProperty("os.version"));
    }

    @Deprecated
    public OS(String name) {
        this(name, System.getProperty("os.version"));
    }

    public OS(String name, String version) {
        this.name = name;
        this.version = version;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        OS os = (OS) o;
        return name.equals(os.name) &&
                version.equals(os.version);
    }

    public boolean equalsIgnoreVersion(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        OS os = (OS) o;
        return name.equals(os.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name, version);
    }

    @NotNull
    public String getVersion() {
        return version;
    }

    @NotNull
    public String getName() {
        return name;
    }
}

File: src/main/kotlin/qtech/utilities/system/OS.kt
package qtech.utilities.system

import java.util.*

class OS constructor(val name: String = System.getProperty("os.name"), val version: String = System.getProperty("os.version")) {

    @Deprecated("")
    constructor(name: String) : this(name, System.getProperty("os.version"))

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val os = other as OS
        return name == os.name && version == os.version
    }

    fun equalsIgnoreVersion(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val os = other as OS
        return name == os.name
    }

    override fun hashCode(): Int {
        return Objects.hash(name, version)
    }
}